gnovm/pkg/doc/doc.go

```diff
@@ -0,0 +1,322 @@
+// Package doc implements support for documentation of Gno packages and realms,
+// in a similar fashion to `go doc`.
+// As a reference, the [official implementation] for `go doc` is used.
+//
+// [official implementation]: https://github.com/golang/go/tree/90dde5dec1126ddf2236730ec57511ced56a512d/src/cmd/doc
+package doc
+
+import (
+	"fmt"
+	"go/ast"
+	"go/doc"
+	"go/token"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.uber.org/multierr"
+)
+
+// DocumentOption is used to pass options to the [Documentable].Document.
+type DocumentOption func(s *documentOptions)
+
+type documentOptions struct {
+	all        bool
+	src        bool
+	unexported bool
+	short      bool
+	w          io.Writer
+}
+
+// WithShowAll shows all symbols when displaying documentation about a package.
+func WithShowAll(b bool) DocumentOption {
+	return func(s *documentOptions) { s.all = b }
+}
+
+// WithSource shows source when documenting a symbol.
+func WithSource(b bool) DocumentOption {
+	return func(s *documentOptions) { s.src = b }
+}
+
+// WithUnexported shows unexported symbols as well as exported.
+func WithUnexported(b bool) DocumentOption {
+	return func(s *documentOptions) { s.unexported = b }
+}
+
+// WithShort shows a one-line representation for each symbol.
+func WithShort(b bool) DocumentOption {
+	return func(s *documentOptions) { s.short = b }
+}
+
+// WithWriter uses the given writer as an output.
+// By default, os.Stdout is used.
+func WithWriter(w io.Writer) DocumentOption {
+	return func(s *documentOptions) { s.w = w }
+}
+
+// Documentable is a package, symbol, or accessible which can be documented.
+type Documentable interface {
+	Document(...DocumentOption) error
+}
+
+type documentable struct {
+	Dir
+	symbol     string
+	accessible string
+	pkgData    *pkgData
+}
+
+func (d *documentable) Document(opts ...DocumentOption) error {
```

Comment:
```suggestion
func (d *documentable) WriteDocument(opts ...DocumentOption) error {
```

or

```suggestion
func (d *documentable) Document(opts ...DocumentOption) (string, error) {
```

Since this package will be called by more than just the CLI, let's make its API more universal.